### PR TITLE
Update skll to 2.0

### PR DESCRIPTION
**Pull request by aloukina**
This branch addresses issues that cause test failures when we move RSMTool to `skll 2.0` (#311, #339). There were no issues related to changes in actual `skll` per se but several issues related to `scipy` update.

* Updated conda environment and changed the channel for conda set up from `desilinguist` to `ets` since `skll` is now in `ets` channel. Updated documentation to reflect this. 
* Added additional handling of cases where n=1 to return `NaN` for  Pearson's `r`. Updated documentation and tests for Pearson's `r`. Added extra tests for situations where we have some groups with n=1. (#343)
* Added additional handling of cases where `n=1` to return `NaN` for `r2`. Updated documentation (#348)
* PCA now uses `min(n_features, n_samples)` to establish the number of components. This is consistent with the default performance in pca code (#347). The documentation and tests were updated accordingly. 
* Re-trained models for rsmpredict using new version of scipy (#344)
* Fixed errors in the list of Utility scripts in documentation (#327)
* Fix coveralls.io integration (#358)

### CODE REVIEW

**Comment by aloukina** [comment]
The remaining PEP8 comments refer to (1) X*X + X*X case where PEP8 doesn't require spaces around high-priority operators. (2) We could shorten variable names to address the other one but I would prefer to keep them as is for consistency.

File: doc/config_rsmtool.rst [hunk 1]
```diff
@@ -165,6 +165,8 @@ RSMTool provides pre-defined sections for ``rsmtool`` (listed below) and, by def
         - the variances; and
         - a Scree plot.
 
+    The analysis keeps all components. The total number of components usually equals the total number of features. In cases where the total number of responses is smaller than the number of features, the number of components is the same as the number of responses.
```
Comment (desilinguist) [inline doc/config_rsmtool.rst]: This paragraph needs to be indented two spaces beyond where the `pca` bullet starts.

File: doc/evaluation.rst [hunk 2]
```diff
@@ -124,7 +124,16 @@ Pearson Correlation coefficient (r)
 
 :math:`r=\displaystyle\frac{\sum_{i=1}^{N}{(H_i-\bar{H})(M_i-\bar{M})}}{\sqrt{\sum_{i=1}^{N}{(H_i-\bar{H})^2} \sum_{i=1}^{N}{(M-\bar{M})^2}}}`
 
-Pearson correlation coefficients is computed using `scipy.stats.pearsonr <https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html>`_. If the variance of human or system scores is ``0`` (all scores are the same), RSMTool returns ``None``.
+Pearson correlation coefficients is computed using `scipy.stats.pearsonr <https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html>`_. 
+
+If the variance of human or system scores is ``0`` (all scores are the same) or only one response is available, RSMTool returns ``None``.
+
+.. note::
+  In `scipy` v1.4.1 and later the implementation uses the following formula:
```
Comment (desilinguist) [inline doc/evaluation.rst]:
```suggestion
  In `scipy` v1.4.1 and later, the implementation uses the following formula: 
```

File: rsmtool/analyzer.py [hunk 3]
```diff
@@ -677,7 +690,9 @@ def metrics_helper(human_scores,
         df = pd.DataFrame({'human': human_scores,
                            'system': system_scores}).dropna(how='any')
 
-        if len(df['human'].unique()) == 1 or len(df['system'].unique()) == 1:
+        if len(df) == 1:
+            correlations = np.nan
+        elif len(df['human'].unique()) == 1 or len(df['system'].unique()) == 1:
```
Comment (desilinguist) [inline rsmtool/analyzer.py]: If both conditions lead to `correlations = np.nan` then why have an `elif`?

**Comment by aloukina** [comment]
Remaining PEP8 comment: https://stackoverflow.com/questions/181530/styling-multi-line-conditions-in-if-statements/31231248#31231248

**Pull request merged**